tests/test_search_pattern.py

```diff
@@ -0,0 +1,187 @@
+    """
+    Initializes a SearchPattern instance.
+    """
+    search_pattern_instance = search_pattern.SearchPattern(
+        camera_fov_forwards=60.0,
+        camera_fov_sideways=30.0,
+        search_height=50.0,
+        search_overlap=0.2,
+        current_position_x=75.0,
+        current_position_y=150.0,
+        distance_squared_threshold=DISTANCE_SQUARED_THRESHOLD,
+        small_adjustment=SMALL_ADJUSTMENT,
+    )
+    yield search_pattern_instance
+
+
+@pytest.fixture()
+def drone_odometry():
+    def create_drone_odometry(pos_x, pos_y, depth):
+        """
+        Creates an OdometryAndTime instance representing the drone's current position.
+        """
+        position = drone_odometry_local.DronePositionLocal.create(pos_y, pos_x, -depth)[
+            1
+        ]
+        orientation = drone_odometry_local.DroneOrientationLocal.create_new(
+            0.0, 0.0, 0.0
+        )[1]
+        odometry_data = drone_odometry_local.DroneOdometryLocal.create(
+            position, orientation
+        )[1]
+
+        result, odometry = odometry_and_time.OdometryAndTime.create(odometry_data)
+        return odometry
+
+    return create_drone_odometry
+
+
+def assert_move_command(move, newPos, target_x, target_y, target_z):
+    assert move[0] == newPos
+    assert (
+        move[1].get_command_type()
+        == decision_command.DecisionCommand.CommandType.MOVE_TO_ABSOLUTE_POSITION
+    )
+    pos = move[1].get_command_position()
+    assert pytest.approx(target_x, 0.1) == pos[0]
+    assert pytest.approx(target_y, 0.1) == pos[1]
+    assert pytest.approx(target_z, 0.1) == pos[2]
+
+
+class TestSearchPattern:
+    """
+    Tests for the SearchPattern class methods
+    """
+
+    def test_width_greater_depth_normal(
+        self, search_pattern_width_greater_depth, drone_odometry
+    ):
+        """
+        Test first 20 positions of search pattern where drone has reached point before being called
+        """
+        coordinates = [
+            [100, 50, True],
+            [123.75, 92.87, False],
+            [124.75, 92.87, True],
+            [142.87, 118.62, False],
+            [142.87, 117.62, True],
+            [142.87, 74.75, True],
+            [168.62, 7.13, False],
+            [167.62, 7.13, True],
+            [124.75, 7.13, True],
+            [57.13, -18.62, False],
+            [57.13, -17.62, True],
+            [57.13, 18.00, True],
+            [57.13, 53.62, True],
+            [57.13, 89.25, True],
+            [57.13, 124.87, True],
+            [31.38, 185.25, False],
+            [32.38, 185.25, True],
+            [68.00, 185.25, True],
+            [103.62, 185.25, True],
+            [139.25, 185.25, True],
+            [174.87, 185.25, True],
+        ]
+
+        for i in range(len(coordinates) - 1):
+            odometry = drone_odometry(coordinates[i][0], coordinates[i][1], 100)
+            move = search_pattern_width_greater_depth.continue_search(odometry)
```

Comment:
use actual_coordinates or another descriptive name here to make it clear this is the output